### DISCUSSION TITLE
Bug with conditionally hiding/showing components?

### DISCUSSION THREAD

**Discussion by sarah-strawberries** [Budibase/budibase#6435] [Help] [answered]
I'm trying to make these "Name(s) of Child(ren)" and "Age(s) of Child(ren)" field components show or hide based on the "Number of Children" field. I added the same two conditions to both the name and age components, shown in the image. But for some reason, only the second condition I add actually does anything. (I tried switching the order of the conditions and got the fields to hide that way, but I can't get them to show up again unless I switch the order of the conditions again.)

I'm aware that when you have a component selected in the editor, it automatically shows up regardless of conditions, but since both the age and name components have the same conditions, only the age component is selected, and the "Number of Children" field is set to 0, it seems like the name component should not be visible right now. Unless only one condition can actually be applied to any given component, in spite of the fact that you can add more than one...(if I use only one condition, the single condition works just fine. So who knows.)

Is this a bug, or am I missing something?
![image](https://user-images.githubusercontent.com/75397080/175106068-60937cad-7c70-4492-93bd-71e90ebd8345.png)

**Comment by sarah-strawberries** [comment]
I found a way around it by changing the second condition to show the component when the field equals 1 instead of not equaling 0. The other way still doesn't work, though.

**Comment by melohagan** [comment, marked as answer]
Hi @sarah-strawberries 

The Show/Hide components will automatically add the inverse scenario. 
So if you have `Show IF condition` then implicitly it will add a `Hide IF NOT condition`
If you have `Hide IF condition` then implicitly it will add a `Show IF NOT condition`

**Reply by sarah-strawberries** [reply]
Aha! Makes sense; thank you!
